agent_memory_server/cli.py

```diff
@@ -0,0 +1,199 @@
+@click.argument("task_path")
+@click.option(
+    "--args",
+    "-a",
+    multiple=True,
+    help="Arguments to pass to the task in the format key=value",
+)
+def schedule_task(task_path: str, args: list[str]):
+    """
+    Schedule a background task by path.
+
+    TASK_PATH is the import path to the task function, e.g.,
+    "agent_memory_server.long_term_memory.compact_long_term_memories"
+    """
+    import asyncio
+
+    from docket import Docket
+
+    # Parse the arguments
+    task_args = {}
+    for arg in args:
+        try:
+            key, value = arg.split("=", 1)
+            # Try to convert to appropriate type
+            if value.lower() == "true":
+                task_args[key] = True
+            elif value.lower() == "false":
+                task_args[key] = False
+            elif value.isdigit():
+                task_args[key] = int(value)
+            elif value.replace(".", "", 1).isdigit() and value.count(".") <= 1:
+                task_args[key] = float(value)
+            else:
+                task_args[key] = value
+        except ValueError:
+            click.echo(f"Invalid argument format: {arg}. Use key=value format.")
+            sys.exit(1)
+
+    async def setup_and_run_task():
+        redis = await get_redis_conn()
+        await ensure_search_index_exists(redis)
+
+        # Import the task function
+        module_path, function_name = task_path.rsplit(".", 1)
+        try:
+            module = importlib.import_module(module_path)
+            task_func = getattr(module, function_name)
+        except (ImportError, AttributeError) as e:
+            click.echo(f"Error importing task: {e}")
+            sys.exit(1)
+
+        # Initialize Docket client
+        async with Docket(
+            name=settings.docket_name,
+            url=settings.redis_url,
+        ) as docket:
+            click.echo(f"Scheduling task {task_path} with arguments: {task_args}")
+            await docket.add(task_func)(**task_args)
+            click.echo("Task scheduled successfully")
+
+    asyncio.run(setup_and_run_task())
+
+
+@cli.command()
+@click.option(
+    "--concurrency", default=10, help="Number of tasks to process concurrently"
+)
+@click.option(
+    "--redelivery-timeout",
+    default=30,
+    help="Seconds to wait before redelivering a task to another worker",
+)
+def task_worker(concurrency: int, redelivery_timeout: int):
+    """
+    Start a Docket worker using the Docket name from settings.
+
+    This command starts a worker that processes background tasks registered
+    with Docket. The worker uses the Docket name from settings.
+    """
+    import asyncio
+
+    from docket import Worker
+
+    if not settings.use_docket:
+        click.echo("Docket is disabled in settings. Cannot run worker.")
+        sys.exit(1)
+
+    asyncio.run(
+        Worker.run(
+            docket_name=settings.docket_name,
+            url=settings.redis_url,
+            name="agent-memory-worker",
```

Comment:
There's no harm in using a hard-coded name, but in cases where a person was running multiple task workers, they'd all report their status and health under the same keys and might scramble the `docket snapshot` a little bit.  I'd just remove this and let docket generate the default name, which is like `hostname#pid`.

```suggestion
```

Reply:
Ahh, yeah, good point!